src/sentry/integrations/jira/client.py

```diff
@@ -224,3 +225,6 @@ def get_field_autocomplete(self, name, value):
         return self.get_cached(
             self.AUTOCOMPLETE_URL, params={"fieldName": jql_name, "fieldValue": value}
         )
+
+    def get_project_statuses(self, project_id: str) -> dict[str, Any]:
+        return dict(self.get_cached(self.STATUS_SEARCH_URL, params={"projectId": project_id}))
```

Comment:
do we need to paginate through this endpoint? or set `maxResults` to something big? https://developer.atlassian.com/cloud/jira/platform/rest/v2/api-group-status/#api-rest-api-2-statuses-search-get

Reply:
`maxResults` is set to 200 by default so we should be good in most cases. We'll only need to handle pagination if we decide to implement the bulk lookup.